compiler/src/test/java/com/squareup/anvil/compiler/ModuleMergerRepeatableTest.kt

```diff
@@ -0,0 +1,347 @@
+package com.squareup.anvil.compiler
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.anvil.annotations.MergeComponent
+import com.squareup.anvil.annotations.MergeSubcomponent
+import com.squareup.anvil.annotations.compat.MergeModules
+import com.squareup.anvil.compiler.internal.testing.anyDaggerComponent
+import com.squareup.anvil.compiler.internal.testing.daggerComponent
+import com.squareup.anvil.compiler.internal.testing.withoutAnvilModule
+import org.junit.Assume.assumeTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Parameterized.Parameters
+import kotlin.reflect.KClass
+
+@RunWith(Parameterized::class)
+class ModuleMergerRepeatableTest(
+  private val annotationClass: KClass<*>
+) {
+
+  private val annotation = "@${annotationClass.simpleName}"
+  private val import = "import ${annotationClass.java.canonicalName}"
+
+  companion object {
+    @Parameters(name = "{0}")
+    @JvmStatic fun annotationClasses(): Collection<Any> {
+      return buildList {
+        add(MergeComponent::class)
+        if (isFullTestRun()) {
+          add(MergeSubcomponent::class)
+          add(MergeModules::class)
+        }
+      }
+    }
+  }
+
+  @Test fun `duplicate scopes are an error`() {
+    assumeIrBackend()
```

Comment:
It's quite unfortunate that we can't test the old backend. The old backend is still used for the stub generating task. I added some integration tests for that.

Reply:
> It's quite unfortunate that we can't test the old backend. The old backend is still used for the stub generating task. I added some integration tests for that.

Could you explain this more? Would we not be using the old backend if we didn't have `assumeIrBackend()` at the start of this test?

Reply:
Without this and using the old backend you always get this error:
```
Repeatable annotations with non-SOURCE retention are only supported starting from Kotlin 1.6
```
The old backend doesn't support language features from Koltin 1.6 and newer. But for some reason it still somehow works for the stub generating task and this check seems to be disabled. Jetbrains is working to remove the old backend for the stub generating task. See https://github.com/square/anvil/issues/422#issuecomment-989458608